orchagent/bufferorch.cpp

```diff
@@ -48,11 +48,7 @@ BufferOrch::BufferOrch(DBConnector *applDb, DBConnector *confDb, DBConnector *st
     m_flexCounterTable(new ProducerTable(m_flexCounterDb.get(), FLEX_COUNTER_TABLE)),
     m_flexCounterGroupTable(new ProducerTable(m_flexCounterDb.get(), FLEX_COUNTER_GROUP_TABLE)),
     m_countersDb(new DBConnector("COUNTERS_DB", 0)),
-    m_stateBufferMaximumValueTable(stateDb, STATE_BUFFER_MAXIMUM_VALUE_TABLE),
-    m_ingressZeroBufferPool(SAI_NULL_OBJECT_ID),
-    m_egressZeroBufferPool(SAI_NULL_OBJECT_ID),
-    m_ingressZeroPoolRefCount(0),
-    m_egressZeroPoolRefCount(0)
```

Comment:
Is there any reason why we don't want use the same egress zero pool for pfcwd and reclaim buffer? @stephenxs

Reply:
yes @neethajohn 
1. Currently, Only [ingress zero buffer pool](https://github.com/Azure/sonic-buildimage/blob/master/platform/mellanox/zero_profiles.j2#L3) is required for reclaiming PG. For the queues, we do not need an egress zero buffer pool because the zero profiles are applied based on normal egress buffer pools. Now that we will no longer apply zero buffer profile on the ingress side, the mechanism to share the zero buffer between reclaiming buffer and PFC WD isn't really required.
2. There is a logic in PFC WD to query the currently applied buffer profile on PG/queue from SAI before applying zero buffer profile. The motivation is to restore the original buffer profile once PFC storm has gone. However, this mechanism introduces an [issue#2231](https://github.com/Azure/sonic-swss/issues/2231) on another vendor's platform while upgrading the switch from an old version to a new one via warm reboot. This is because  PFC storm can happen before APPLY_VIEW during warm reboot when `GET` operation is prohibitted.  There has been a WA in sairedis to make `GET` works in this scenario. However, the zero buffer pool/profile that was applied by PFC WD differs from that created for reclaiming buffer, which fails the sairedis WA. (It is still very strange that the mechanism had existed for years but the issue was reported only last year). 

So we decide to revert it for now.
If we still want this mechanism to share zero buffer between reclaiming buffer and PFC WD, we may want to enhance PFC WD and buffer orch a bit more:
- buffer orch to store the buffer profiles currently applied on the queue/PG
- PFC WD not to query SAI for the currently applied buffer profiles

Let me know in case it is not clear to you

Reply:
Thanks Stephen for the detailed explanation!